### ISSUE TITLE
ArgumentError with timex 3.7.2 and elixir 1.11.3

### ISSUE THREAD

**Issue by lanarchyste** [bitwalker/timex#645]
Hi,

I have an exception with timex 3.7.2 when I use the shift method :

> % ArgumentError 
{message: "cannot add 86400000000 microsecond to #DateTime<2021-03-19 14:46:26.411962+01:00 +01 Etc/UTC+1> (with time zone database Tzdata.TimeZoneDatabase), reason: :time_zone_not_found"}, 
[{DateTime,: add,4,[file: 'lib/calendar/datetime.ex', line: 1338]}, 
{Timex.Protocol.DateTime,: shift,2,[file: 'lib/datetime/datetime.ex', line: 420]}

I am using tzdata 1.1.0

I have no issue with timex 3.7.0 and the same version of tzdata

**Comment by bitwalker** [comment]
I’ll take a look ASAP

**Comment by bitwalker** [comment]
By the way, could you give me the arguments passed?

**Comment by wearethefoos** [comment]
We see this issue as well. You can reproduce this by this part of one of our tests:

```elixir
start_date = Timex.format!(
            Timex.add(
              Timex.set(Timex.beginning_of_week(Timex.now("Europe/Amsterdam")), hour: 10),
              Timex.Duration.from_weeks(4)
            ),
            "{ISO:Extended}"
          )

parsed_date = Timex.parse!(start_date, "{ISO:Extended}")

end_date = Timex.add(parsed_date, Timex.Duration.from_days(7))
```

**Comment by bitwalker** [comment]
This is fixed in 3.7.3, release will be out in a few

**Comment by wearethefoos** [comment]
Thank you!
